### PR TITLE
Added search-sort-filter feature

### PR DESCRIPTION
**Pull request by nikhil-sharma-nks**
changelogs

1. Added search functionality
2. Added Sort functionality
3. Added Filter functionality

filter

### CODE REVIEW

File: src/components/FilterSearch/FilterSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  ];
+
+  const checkIfChecked = (label) => {
+    return selectedTags.includes(label);
+  };
+  const handleLabelClick = (label, isChecked) => {
+    if (isChecked) {
+      setSelectedTags((s) => {
+        noteDispatch({ type: 'FILTER_TAGS', payload: [...s, label] });
+        return [...s, label];
```
Comment (atul-gairola) [inline src/components/FilterSearch/FilterSearch.jsx:31]: good use of updater function

File: src/components/FilterSearch/FilterSearch.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    } else {
+      const newSet = new Set(selectedTags);
+      newSet.delete(label);
+      setSelectedTags([...newSet]);
+      noteDispatch({ type: 'FILTER_TAGS', payload: [...newSet] });
```
Comment (atul-gairola) [inline src/components/FilterSearch/FilterSearch.jsx:37]: good use of set

File: src/components/FilterSearch/FilterSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    }),
+    singleValue: (provided, state) => ({
+      ...provided,
+      fontSize: state.selectProps.myFontSize,
+    }),
+  };
+  const toggleFilterModal = () => setIsFilterModalOpen((prev) => !prev);
+  return (
+    <>
+      <div className='search-filter-container'>
```
Comment (atul-gairola) [inline src/components/FilterSearch/FilterSearch.jsx:81]: Filter and search could've been 2 separate components

**Review by atul-gairola — APPROVED** [review]
Looks good to me

**Review by akash-1318 — COMMENTED** [review]
Code is really clean with proper formatting and nicely use of useReducer and use context hooks. Good work.

**Pull request merged**